engine/classes/Elgg/Filesystem/Adapter/AwsS3.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace Elgg\Filesystem\Adapter;
+use Gaufrette\Filesystem;
+use Gaufrette\File;
+use Gaufrette\Adapter\AwsS3 as GaufretteAwsS3;
+use Aws\S3\S3Client;
+
+class AwsS3 extends Filesystem implements Adapter {
+	private $config;
+	public function __construct(array $config) {
```

Comment:
The work has to be done somewhere, and I thought it was cleaner to do it here than in ElggFile. Doing it here also keeps Gaufrette further away from the public API.